docs/usage/types/bytesize.md

```diff
@@ -1,32 +1,49 @@
 # Bytes and ByteSize
 
 `bytes`
-: `bytes` are accepted as-is, `bytearray` is converted using `bytes(v)`, `str` are converted using `v.encode()`,
-  and `int`, `float`, and `Decimal` are coerced using `str(v).encode()`
+: `bytes` are accepted as-is, `bytearray` is converted using `bytes(v)`, and `str` is converted using `v.encode()`.
 
 `SecretBytes`
-: bytes where the value is kept partially secret; see [Secrets](secrets.md)
-
-`conbytes`
-: type method for constraining bytes
+: Like `bytes`, but where the value is kept partially secret; see [Secrets](secrets.md).
 
 `ByteSize`
-: converts a bytes string with units to bytes
+: Converts a string representing a number of bytes with units (such as `'1KB'` or `'11.5MiB'`) into an integer.
+
+`conbytes`
+: A function which returns a type that can be used to add constraints to `bytes`.
 
 ### Arguments to `conbytes`
 The following arguments are available when using the `conbytes` type function
 
-- `strip_whitespace: bool = False`: removes leading and trailing whitespace
-- `to_upper: bool = False`: turns all characters to uppercase
-- `to_lower: bool = False`: turns all characters to lowercase
 - `min_length: int = None`: minimum length of the byte string
 - `max_length: int = None`: maximum length of the byte string
 - `strict: bool = False`: controls type coercion
 
+However, for the sake of improved integration with type checkers, we now discourage the use of `conbytes` and other
```

Comment:
"discouraging" seems ambiguous here. Are those deprecated or not? If they aren't deprecated then we just should support them. Are we going to deprecate those in one of the following releases?